### ISSUE TITLE
[User Feedback] - Issue Trace Timeline

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry#68481]
tl;dr: Add the new trace timeline to User Feedback details

A new component was built for the Issue Details page called [Issue Trace Timeline](https://sentry.io/changelog/issues-trace-timeline/). Previously known as Trace Navigator

<img width="905" alt="image" src="https://github.com/getsentry/sentry/assets/1633368/cac77294-32e7-4e2f-9708-8cafa6b1e5b2">

With the JS SDK v8, each new page load or navigation will create a new trace. That means that if a user submits a User Feedback through the widget, that feedback will be using the same trace-id as other relevant data during that last page load. Including client side errors, as well as calls to the backend that might have be the culprit of that error.

With the JS SDK v7 though, there is a global trace id that wasn't used during page load requests that resulted in HTTP calls to the backend. This means that User Feedbacks from the widget are not likely to be connected to any other telemetry.

Additionally, feedback received from the legacy API (that only takes name/email/comments) doesn't have a trace id. They are connected to an error though (through `event_id`) so we potentially leverage that trace-id. We can do so by coping the `trace_id` from the [event linked in the shim code](https://github.com/getsentry/sentry/blob/1d5adc7369f53bf2e9396b1aba8a0e21f67cd5f1/src/sentry/feedback/usecases/create_feedback.py#L290-L302).

To verify the data, we can use `?dataset=issuePlatform` on Discover. For example [this query](https://sentry.sentry.io/discover/homepage/?dataset=issuePlatform&field=user.display&field=timestamp&field=trace&name=All+Events&project=11276&query=title%3A%22User+Feedback%22&sort=-timestamp&statsPeriod=14d&yAxis=count%28%29). Once v8 is installed in Sentry we'll be able to confirm what data would be visible in the component.

Also note the component might need to be adjusted to show User Feedback. It was designed for Issue Details so it shows other error events only (no spans).

<details>
<summary>Additional context</summary>

## Spans on trace navigator

<img width="786" alt="image" src="https://github.com/getsentry/sentry/assets/1633368/72d828f2-711e-4a82-bb2c-306a6500a070">

## Trace-id behavior in JS SDK v7 vs v8

<img width="1107" alt="image" src="https://github.com/getsentry/sentry/assets/1633368/cb8e543e-0876-468f-ab5f-509e62ade260">

<img width="1113" alt="image" src="https://github.com/getsentry/sentry/assets/1633368/ff2c8e73-9cf8-4684-9cb4-8ebfde2afc62">
<img width="1108" alt="image" src="https://github.com/getsentry/sentry/assets/1633368/ae58b47e-70f0-416f-9d72-1e9ff549f1f1">

</details>

**Comment by bruno-garcia** [comment]
Depends on TwP working properly:

* https://github.com/getsentry/sentry-javascript/pull/11561

**Comment by aliu39** [comment]
@bruno-garcia 

> To verify the data, we can use ?dataset=issuePlatform on Discover. For example [this query](https://sentry.sentry.io/discover/homepage/?dataset=issuePlatform&field=user.display&field=timestamp&field=trace&name=All+Events&project=11276&query=title%3A%22User+Feedback%22&sort=-timestamp&statsPeriod=14d&yAxis=count%28%29). Once v8 is installed in Sentry we'll be able to confirm what data would be visible in the component.

Can you clarify what is meant by "verify the data" here? Also, do you know when v8 will be installed?

**Comment by jas-kas** [comment]
@aliu39 the v8 SDK has been installed for Sentry :)

by verify the data, i think bruno means to click into the trace IDs associated with the feedback events and confirm if the same issues in those traces, show up in the issue trace timeline

![Image](https://github.com/user-attachments/assets/b0068cc6-290c-4e25-a145-af58ef13c3af)

**Comment by aliu39** [comment]
> Also note the component might need to be adjusted to show User Feedback. It was designed for Issue Details so it shows other error events only (no spans).

I need to find an example, but looking over the code, I believe the component already shows user feedback. UF is an Issue Platform event and [this query for timeline events](https://github.com/getsentry/sentry/blob/0f8052846a3eb1c7f05f71433385ddde24c72667/static/app/views/issueDetails/traceTimeline/useTraceTimelineEvents.tsx#L68-L96) doesn't filter by category. So as long as a UF has a trace id, it'll show up

**Comment by aliu39** [comment]
Closed by https://github.com/getsentry/sentry/pull/76858

**Comment by aliu39** [comment]
Still need to GA this

**Comment by aliu39** [comment]
Amplitude dashboard: https://app.amplitude.com/analytics/sentry/dashboard/igfiqgc0?source=copy+url
